src/main/java/redis/clients/jedis/JedisPubSub.java

```diff
@@ -15,11 +15,9 @@
 import redis.clients.jedis.exceptions.JedisException;
 import redis.clients.util.SafeEncoder;
 
-public abstract class JedisPubSub {
+public abstract class JedisPubSub extends BinaryJedisPubSub {
```

Comment:
@marcosnils  @HeartSaVioR  I have removed the subclassing. And added support to BinaryJedisPubSub.java. Submitted new changes to pull request. Please take a look.